funsor/terms.py

```diff
@@ -30,15 +30,20 @@
 from funsor.six import getargspec, singledispatch
 
 
+def _cons_cache_key(*args):
```

Comment:
nit: I believe it is slightly cheaper to pass `args` rather than `*args`. I might even inline this one-line function.